src/elements/stepper/stepper/stepper.ts

```diff
@@ -0,0 +1,307 @@
+
+/**
+ * Provides a structured, step-by-step workflow for user interactions.
+ * @slot - Provide a `sbb-expansion-panel-header` and a `sbb-expansion-panel-content` to the stepper.
+ * @slot step-label - Use this slot to provide an `sbb-step-label`.
+ * @slot step - Use this slot to provide an `sbb-step`.
+ */
+@customElement('sbb-stepper')
+export class SbbStepperElement extends LitElement {
+  public static override styles: CSSResultGroup = style;
+
+  /** If set to true, only the current and previous labels can be clicked and selected. */
+  @property({ type: Boolean }) public linear = false;
+
+  /** Overrides the behaviour of `orientation` property. */
+  @property({ attribute: 'horizontal-from', reflect: true })
+  public get horizontalFrom(): SbbHorizontalFrom | undefined {
+    return this._horizontalFrom;
+  }
+  public set horizontalFrom(value: SbbHorizontalFrom) {
+    this._horizontalFrom = breakpoints.includes(value) ? value : undefined;
+    if (this._horizontalFrom && this._loaded) {
+      this._checkOrientation();
+    }
+  }
+  private _horizontalFrom?: SbbHorizontalFrom | undefined;
+
+  /** Steps orientation, either horizontal or vertical. */
+  @property({ reflect: true })
+  public orientation: SbbOrientation = 'horizontal';
+
+  /** The currently selected step. */
+  @property({ attribute: false })
+  public set selected(step: SbbStepElement) {
+    if (this._loaded) {
+      this._select(step);
+    }
+  }
+  public get selected(): SbbStepElement | undefined {
+    return this.querySelector?.<SbbStepElement>('sbb-step[data-selected]') ?? undefined;
+  }
+
+  /** The currently selected step index. */
+  @property({ attribute: 'selected-index', type: Number })
+  public set selectedIndex(index: number) {
+    if (this._loaded) {
+      this._select(this.steps[index]);
+    }
+  }
+  public get selectedIndex(): number | undefined {
+    return this.selected ? this.steps.indexOf(this.selected) : undefined;
+  }
+
+  /** The steps of the stepper. */
+  public get steps(): SbbStepElement[] {
+    return Array.from(this.querySelectorAll?.('sbb-step') ?? []);
+  }
+
+  private get _enabledSteps(): SbbStepElement[] {
+    return this.steps.filter((s) => !s.label?.hasAttribute('disabled'));
+  }
+
+  /** Selects the next step. */
+  public next(): void {
+    if (this.selectedIndex !== undefined) {
+      this._select(this.steps[this.selectedIndex + 1]);
+    }
+  }
+
+  /** Selects the previous step. */
+  public previous(): void {
+    if (this.selectedIndex !== undefined) {
+      this._select(this.steps[this.selectedIndex - 1]);
+    }
+  }
+
+  /** Resets the form in which the stepper is nested or every form of each step, if any. */
+  public reset(): void {
+    const closestForm = this.closest('form');
+    if (closestForm) {
+      closestForm.reset();
+    } else {
+      this.querySelectorAll('form').forEach((form) => form.reset());
+    }
+    this.selectedIndex = 0;
+    this.selected?.label?.focus();
```

Comment:
Briefly discussed with Jeri. @dauriamarco What do you think?
```suggestion
    if (document.activeElement?.closest('sbb-stepper') === this) {
      this.selected?.label?.focus();
    }
```

Reply:
works for me 👍